pkg/controllermanager/controller/seed/seed_reconcile.go

```diff
@@ -157,25 +155,18 @@ func (r *reconciler) syncGardenSecrets(ctx context.Context, gardenClient kuberne
 		fns = append(fns, func(ctx context.Context) error {
 			seedSecret := &corev1.Secret{
 				ObjectMeta: metav1.ObjectMeta{
-					Annotations: secret.Annotations,
-					Labels:      secret.Labels,
-					Name:        secret.Name,
-					Namespace:   namespace.Name,
+					Name:      secret.Name,
+					Namespace: namespace.Name,
 				},
-				Data: secret.Data,
 			}
-			// TODO: use controllerutil.CreateOrUpdate once cache is graduated to beta https://github.com/gardener/gardener/issues/2822
-			if err := gardenClient.Client().Create(ctx, seedSecret); err != nil {
-				if !apierrors.IsAlreadyExists(err) {
-					return err
-				}
-				_, err2 := kutil.TryUpdateSecret(ctx, gardenClient.Kubernetes(), retry.DefaultRetry, seedSecret.ObjectMeta, func(obj *corev1.Secret) (*corev1.Secret, error) {
```

Comment:
Can this function be removed now?

Reply:
Yes, but let's do it with some other PR.

Reply:
Hm, why? You are already touching the code, so can't you just remove it now that nobody uses it anymore?

Reply:
Because it now needs some more 30 minutes just to re-build and verify the PR. And this is just in case the CI does not fail.

Anyway, I have applied the change.